src/interceptors/ClientRequest/index.ts

```diff
@@ -103,6 +106,11 @@ export const interceptClientRequest: Interceptor = (middleware) => {
   return () => {
     debug('restoring patches...')
 
+    //restore the original ClientReqest if it was overridden
+    if (originalClientRequest) {
```

Comment:
I think this condition is redundant. This is the restore function that a module interceptor returns, and it must always restore any patched references. 

Do you see any harm in dropping the condition? Can it be that we restore the `http.ClientRequest` to a patched version due to some concurrency or multiple instances of `RequestInterceptor`?

Reply:
Yes you are right we can drop

Reply:
I believe even if there's a multiple instances issue, calling a restore function must still restore what it patched. Then the issue would be in multiple instances, while the behavior of restore would remain as expected.

Reply:
Maybe I don't understand your point. If I restore an instance then when a new request is intercepted `ClientRequest` will be patch again